File: Coursera/BinarySearch.java
class BinarySearch{
    public static int BinarySearch(int[] a, int key)
    {
        int lo = 0, high = a.length - 1; 
        while(lo < high)
        {
            int mid = (lo - (high - lo) / 2); 
            if(key < a[mid]) hi = mid - 1; 
            else if (key > a[mid]) lo = mid + 1; 
            else{
                return mid; 
            }
        }
        return -1; 
    }
}

File: Udemy/binary_search.cpp
#include <iostream>
using namespace std;

int binarySearch(int nums[], int low, int high, int num){
    if(low > high) return -1; 

    int mid = low + (high - low) / 2; 
    if(nums[mid] == num) return mid;
    if(num > nums[mid]) return binarySearch(nums, mid+1, high, num);
    else return binarySearch(nums, low, mid-1, num); 
}